### PR TITLE
vectorstores: pgvector add option WithConn

### PR DESCRIPTION
**Pull request by Abirdcfly**
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

**Review by tmc — COMMENTED** [review]
looking good but please address lint issues

File: vectorstores/pgvector/pgvector.go [hunk 1]
```diff
@@ -55,33 +55,41 @@ func New(ctx context.Context, opts ...Option) (Store, error) {
 	if err != nil {
 		return Store{}, err
 	}
-	store.conn, err = pgx.Connect(ctx, store.postgresConnectionURL)
```
Comment (tmc) [inline vectorstores/pgvector/pgvector.go:58]: This constructor should return a pointer to Store

Comment (Abirdcfly) [reply vectorstores/pgvector/pgvector.go:58]: Please correct me if I am mistaken, but are you saying that the `func New(ctx context.Context, opts ...Option) (Store, error)` on line 53 should return a `*Store`?

Comment (tmc) [reply vectorstores/pgvector/pgvector.go:58]: Yes, (*Store, error), but we can correct that in another PR

**Pull request merged**